hazelcast/src/main/java/com/hazelcast/util/VersionCheck.java

```diff
@@ -118,33 +109,33 @@ private void doCheck(Node hazelcastNode, String version, boolean isEnterprise) {
             IOUtil.closeResource(is);
         }
 
-        String urlStr = "http://versioncheck.hazelcast.com/version.jsp?version=" + version
-                + "&m=" + hazelcastNode.getLocalMember().getUuid()
-                + "&e=" + isEnterprise
-                + "&l=" + toMD5String(hazelcastNode.getConfig().getLicenseKey())
-                + "&p=" + downloadId
-                + "&c=" + toMD5String(hazelcastNode.getConfig().getGroupConfig().getName())
-                + "&crsz=" + convertToLetter(hazelcastNode.getClusterService().getMembers().size())
-                + "&cssz=" + convertToLetter(hazelcastNode.clientEngine.getClientEndpointCount());
-
+        //Calculate native memory usage from native memory config
+        NativeMemoryConfig memoryConfig = hazelcastNode.getConfig().getNativeMemoryConfig();
+        long totalNativeMemorySize = hazelcastNode.getClusterService().getSize()
+                * memoryConfig.getSize().bytes();
+        String nativeMemoryParameter = (isEnterprise)
+                ? Long.toString(MemoryUnit.BYTES.toGigaBytes(totalNativeMemorySize)) : "0";
+        //Calculate connected clients to the cluster.
+        Map<ClientType, Integer> clusterClientStats = hazelcastNode.getClusterService().getConnectedClientStats();
+
+        UrlActionParameterCreator parameterCreator = new UrlActionParameterCreator();
+        parameterCreator.addParam("version", version);
+        parameterCreator.addParam("m", hazelcastNode.getLocalMember().getUuid());
+        parameterCreator.addParam("e", Boolean.toString(isEnterprise));
+        parameterCreator.addParam("l", MD5Util.toMD5String(hazelcastNode.getConfig().getLicenseKey()));
+        parameterCreator.addParam("p", downloadId);
+        parameterCreator.addParam("c", MD5Util.toMD5String(hazelcastNode.getConfig().getGroupConfig().getName()));
+        parameterCreator.addParam("crsz", convertToLetter(hazelcastNode.getClusterService().getMembers().size()));
+        parameterCreator.addParam("cssz", convertToLetter(hazelcastNode.clientEngine.getClientEndpointCount()));
+        parameterCreator.addParam("hdgb", nativeMemoryParameter);
+        parameterCreator.addParam("ccpp", Integer.toString(clusterClientStats.get(ClientType.CPP)));
+        parameterCreator.addParam("cdn", Integer.toString(clusterClientStats.get(ClientType.CSHARP)));
+        parameterCreator.addParam("cjv", Integer.toString(clusterClientStats.get(ClientType.JAVA)));
```

Comment:
You can calculate number of enterprise client from isEnterprise parameter, because with the new license ee clients will only os clients will only connect to os clients.  There is a new pr which only contains changes about call home. https://github.com/hazelcast/hazelcast/pull/5137.